src/actions/motion.ts

```diff
@@ -447,6 +447,73 @@ class CommandPreviousSearchMatch extends BaseMovement {
   }
 }
 
+enum VisualMark {
+  SelectionStart,
+  SelectionEnd,
+}
+abstract class MarkMovementVisual extends BaseMovement {
+  isJump = true;
+  registerMode: RegisterMode;
+  modes = [Mode.Normal];
+  mark: VisualMark;
+
+  private startOrEnd(lastVisualSelection): Position {
+    // marks from vimstate are sorted by direction of selection (moving forward vs backwards).
+    // must sort to document order
+    let [start, end] = sorted(lastVisualSelection.start, lastVisualSelection.end);
+    return this.mark === VisualMark.SelectionStart ? start : end;
+  }
+
+  private inLineCorrection(position: Position): Position {
+    // for ' mark, we must go to BOL.
+    // for `> mark, we must correct by one char left
+    return this.registerMode === RegisterMode.LineWise
+      ? position.getLineBeginRespectingIndent()
+      : this.mark === VisualMark.SelectionStart
+      ? position
+      : position.getLeft();
+  }
```

Comment:
For end of visual selection, we have to correct one char to the left.
Is this linked to
https://github.com/VSCodeVim/Vim/pull/5049#issue-452966423 ?

Reply:
Yeah I think so. It's fine for now